odl/discr/discr_ops.py

```diff
@@ -54,74 +61,81 @@ def finite_diff(f, out=None, axis=0, dx=1.0, edge_order=2,
-        Default distance: 1.0
-    edge_order : {1, 2}, optional
-        First-order accurate differences (1) can be used at the boundaries
-        if no zero padding is used. Default edge order: 2
-    zero_padding : `bool`, optional
-        Implicit zero padding. Assumes values outside the domain of ``f`` to be
-        zero. Default: `False`
+        Scalar specifying the distance between sampling points along ``axis``
     method : {'central', 'forward', 'backward'}, optional
-        The method that should be used.
+        Finite difference method which is used in the interior of the domain
+         of ``f``.
+    padding : `float` or 'edge', optional
+        `float` : Implicit padding. Assumes indices outside the domain of
+            ``f`` to have the value ``padding``. If `None` forward or
+              backward differences are used at the boundary instead of zero
+              padding.
+        'edge' : Replicate values at the boundary. Thus forward or
+        backward difference become zero on the endpoints.
+    edge_order : {1, 2}, optional
+        Edge-order accuracy at the boundaries if no padding is used. If
+        `None` the edge-order accuracy at endpoints corresponds to the
+        accuracy in the interior.
 
     Returns
     -------
     out : `numpy.ndarray`
-        N-dimensional array of the same shape as ``f``. If ``out`` was
-            provided, the returned object is a reference to it.
+        N-dimensional array of the same shape as ``f``. If ``out`` is
+        provided, the returned object is a reference to it.
 
     Examples
     --------
     >>> f = np.array([ 0., 1., 2., 3., 4., 5., 6., 7., 8., 9.])
+
     >>> finite_diff(f)
     array([ 1.,  1.,  1.,  1.,  1.,  1.,  1.,  1.,  1.,  1.])
-    >>> finite_diff(f, axis=0, dx=1.0, edge_order=2, zero_padding=False)
+
+    Without arguments the above defaults to:
+    >>> finite_diff(f, axis=0, dx=1.0, method='forward', padding=None,
+    ... edge_order=None)
     array([ 1.,  1.,  1.,  1.,  1.,  1.,  1.,  1.,  1.,  1.])
+
     >>> finite_diff(f, dx=0.5)
     array([ 2.,  2.,  2.,  2.,  2.,  2.,  2.,  2.,  2.,  2.])
-    >>> finite_diff(f, zero_padding=True)
-    array([ 0.5,  1. ,  1. ,  1. ,  1. ,  1. ,  1. ,  1. ,  1. , -4. ])
-    >>> finite_diff(f, zero_padding=False, edge_order=1)
-    array([ 1.,  1.,  1.,  1.,  1.,  1.,  1.,  1.,  1.,  1.])
-    >>> out = finite_diff(f)
-    >>> out is finite_diff(f)
-    False
+    >>> finite_diff(f, padding=0)
+    array([ 1.,  1.,  1.,  1.,  1.,  1.,  1.,  1.,  1., -9.])
+
+    Central differences and different edge orders
+    >>> finite_diff(1/2*f**2, method='central')
+    array([-0.,  1.,  2.,  3.,  4.,  5.,  6.,  7.,  8.,  9.])
+    >>> finite_diff(1/2*f**2, method='central', edge_order=1)
+    array([ 0.5,  1. ,  2. ,  3. ,  4. ,  5. ,  6. ,  7. ,  8. ,  8.5])
+
+    In-place evaluation:
+    >>> out = f.copy()
     >>> out is finite_diff(f, out)
```

Comment:
Use `out` as keyword parameter. That makes it robust regarding changes in the order of parameters (see above).